x-pack/plugins/security_solution/scripts/endpoint/common/delete_all_endpoint_data.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Client, estypes } from '@elastic/elasticsearch';
+import assert from 'assert';
+import { createEsClient } from './stack_services';
+import { createSecuritySuperuser } from './security_user_services';
+
+export interface DeleteAllEndpointDataResponse {
+  count: number;
+  query: string;
+  response: estypes.DeleteByQueryResponse;
+}
+
+/**
+ * Attempts to delete all data associated with the provided endpoint agent IDs.
+ *
+ * **NOTE:** This utility will create a new role and user that has elevated privileges and access to system indexes.
+ *
+ * @param esClient
+ * @param endpointAgentIds
+ */
+export const deleteAllEndpointData = async (
+  esClient: Client,
+  endpointAgentIds: string[]
+): Promise<DeleteAllEndpointDataResponse> => {
+  assert(endpointAgentIds.length > 0, 'At least one endpoint agent id must be defined');
+
+  const unrestrictedUser = await createSecuritySuperuser(esClient, 'super_superuser');
+  const esUrl = getEsUrlFromClient(esClient);
+  const esClientUnrestricted = createEsClient({
+    url: esUrl,
+    username: unrestrictedUser.username,
+    password: unrestrictedUser.password,
+  });
+
+  const queryString = endpointAgentIds.map((id) => `(${id})`).join(' OR ');
+
+  const deleteResponse = await esClientUnrestricted.deleteByQuery({
+    index: '*,.*',
+    body: {
+      query: {
+        query_string: {
+          query: queryString,
```

Comment:
Not needed now, but perhaps we could do this query more accurate to avoid removing extra things.

Reply:
yeah, I played around with that as well (see a sample query below), but found that as more/other data is added, we are likely to miss it. I wanted to truly try to remove all records that reference the agent ID so that our tests (especially those that use a Real endpoint) clean up properly after themselves. 

Here is the query I initially came up with:

```jsonp
# List all indexes where the agent id has data
GET */_search
{
  "query": {
    "bool": {
      "filter": [
        {
          "bool": {
            "should": [
              {
                "match": {
                  "agent.id": "180f5edb-2f48-4d4b-af97-2434775a0bb9"
                }
              },
              {
                "match": {
                  "agent_id": "180f5edb-2f48-4d4b-af97-2434775a0bb9"
                }
              },
              {
                "match": {
                  "EndpointActions.action_id": "180f5edb-2f48-4d4b-af97-2434775a0bb9"
                }
              },
              {
                "match": {
                  "agents": "180f5edb-2f48-4d4b-af97-2434775a0bb9"
                }
              },
              {
                "match": {
                  "_id": "180f5edb-2f48-4d4b-af97-2434775a0bb9"
                }
              }
            ],
            "minimum_should_match": 1
          }
        }
      ]
    }
  },
  "_source": false,
  "size": 0,
  "aggs": {
    "unique_indexes": {
      "terms": {
        "field": "_index",
        "size": 10000
      }
    }
  }
}
```